code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm

```diff
@@ -1443,3 +1370,77 @@
+	if(!owner || !length(affected_turfs))
+		CRASH("behemoth_area_attack([owner], [length(affected_turfs)], ...) called with improper arguments")
+	if(!isxeno(owner))
+		return
+	var/mob/living/carbon/xenomorph/xeno_owner = owner
+	var/attack_damage = (xeno_owner.xeno_caste.melee_damage * xeno_owner.xeno_melee_damage_modifier) * damage_multiplier
+	for(var/turf/affected_turf AS in affected_turfs)
+		if(isclosedturf(affected_turf))
+			continue
+		new /obj/effect/temp_visual/behemoth/crack(affected_turf)
+		playsound(affected_turf, 'sound/effects/behemoth/seismic_fracture_explosion.ogg', 15)
+		var/attack_vfx = enhanced? /obj/effect/temp_visual/behemoth/seismic_fracture/enhanced : /obj/effect/temp_visual/behemoth/seismic_fracture
+		new attack_vfx(affected_turf, enhanced? FALSE : null)
+		for(var/atom/movable/affected_atom AS in affected_turf)
+			if(isliving(affected_atom))
+				var/mob/living/affected_living = affected_atom
+				if(xeno_owner.issamexenohive(affected_living) || affected_living.stat == DEAD || CHECK_BITFIELD(affected_living.status_flags, INCORPOREAL|GODMODE))
+					continue
+				affected_living.emote("scream")
+				shake_camera(affected_living, 1, 0.8)
+				affected_living.Paralyze(paralyze_duration)
+				affected_living.apply_damage(attack_damage, BRUTE, blocked = MELEE)
+			else if(isearthpillar(affected_atom) || ismecha(affected_atom) || istype(affected_atom, /obj/structure/reagent_dispensers/fueltank))
+				affected_atom.do_jitter_animation()
+				new /obj/effect/temp_visual/behemoth/landslide/hit(affected_atom.loc)
+				playsound(affected_atom.loc, get_sfx("behemoth_earth_pillar_hit"), 40)
+				if(isearthpillar(affected_atom))
+					var/obj/structure/earth_pillar/affected_pillar = affected_atom
+					if(affected_pillar.warning_flashes < initial(affected_pillar.warning_flashes))
+						continue
+					affected_pillar.behemoth_area_attack()
+					var/list/turf/spread_turfs = filled_turfs(affected_pillar.loc, EARTH_PILLAR_SPREAD_RADIUS, include_edge = FALSE, bypass_window = TRUE, projectile = TRUE)
+					var/wind_up_duration = initial(affected_pillar.warning_flashes) * 10
+					do_warning(xeno_owner, spread_turfs, wind_up_duration)
+					addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(behemoth_area_attack), xeno_owner, spread_turfs, enhanced), wind_up_duration)
+					continue
+				if(ismecha(affected_atom))
+					var/obj/vehicle/sealed/mecha/affected_mecha = affected_atom
+					affected_mecha.take_damage(attack_damage * AREA_ATTACK_DAMAGE_MECHA_MODIFIER, MELEE)
+					continue
+				if(istype(affected_atom, /obj/structure/reagent_dispensers/fueltank))
+					var/obj/structure/reagent_dispensers/fueltank/affected_tank = affected_atom
+					affected_tank.explode()
+					continue
+
+/// Warns nearby players, in any way or form, of the incoming ability and the range it will affect.
+/proc/do_warning(owner, list/turf/target_turfs, duration)
```

Comment:
ditto. I know it was already a global but I don't see why, exactly

Reply:
Ditto. This proc exclusively was a suggestion from Tivi back when Behemoth was first added.